DHP-277 Use reactive button to toggle login modal

yonidoronpeters opened the pull request:
## Description
This PR
* Updates login button to be reactive
* Updates button text to "Sign in or create an account" per original requirement
* Adds unit tests to unauthorized view

## Screenshots
Before:
<img width="704" alt="Screen Shot 2022-05-19 at 10 03 33 AM" src="https://user-images.githubusercontent.com/1357047/172094055-e4ae3d87-6fdf-49d3-b930-bae1c168fc3a.png">

After:
<img width="695" alt="Screen Shot 2022-06-05 at 9 17 53 PM" src="https://user-images.githubusercontent.com/1357047/172094039-8b4f6771-54b3-4004-857d-110ea3113f2a.png">

yonidoronpeters committed:
DHP-277 Use reactive button to toggle login modal

jerekshoe commented on src/applications/dhp-connected-devices/components/UnauthenticatedPageContent.jsx:
```diff
@@ -1,7 +1,12 @@
+import PropTypes from 'prop-types';
 
-export const UnauthenticatedPageContent = () => {
+export const UnauthenticatedPageContent = toggleLoginModal => {
```
This should be:
```js
export const UnauthenticatedPageContent = ({ toggleLoginModal }) => {
```

jerekshoe commented on src/applications/dhp-connected-devices/components/UnauthenticatedPageContent.jsx:
```diff
@@ -1,7 +1,12 @@
-import React from 'react';
```
Sorry to butt in here, but I saw your request in platform support so I figured I'd take a peek and see if I could come up with anything. From what I can tell there are 2 issues.
1) My comment on L6
2) Your import for this component is incorrect here:
https://github.com/department-of-veterans-affairs/vets-website/blob/5d3affd8d58956e233feae4e0e8701e20ba38316/src/applications/dhp-connected-devices/containers/App.jsx#L3
It should be: 
```js
import UnauthenticatedPageContent from '../components/UnauthenticatedPageContent';
```
You need the connected version of the component so that your action creator(s) from `mapDispatchToProps` get passed into the component as props

yonidoronpeters replied:
OMG @jerekshoe thank you so so much for the detailed explanation and for catching those issues. I was going to struggle with this for much much longer without your help. You rock!!

jerekshoe replied:
I’ve definitely made both of these mistakes in the past so I know how hard they can be to catch. I’m glad that you got it working!

jerekshoe commented on src/applications/dhp-connected-devices/components/UnauthenticatedPageContent.jsx:
```diff
@@ -16,10 +21,23 @@ export const UnauthenticatedPageContent = () => {
 };
+
+const mapDispatchToProps = dispatch => ({
+  toggleLoginModal: open => dispatch(toggleLoginModalAction(open)),
```
This is just a suggestion, but since you know that you are only going to be using `true` as the argument for this function, you could simply write this as:
```js
toggleLoginModal: () => dispatch(toggleLoginModalAction(true)),
```

This would allow you to replace L20 above with:
```js
<button type="button" className="usa-button" onClick={toggleLoginModal}>
```
and you would could then remove `onSignInClicked`

yonidoronpeters committed:
DHP-277 Fix sign in modal button by
1. Fixing props in functional component
2. Fixing import for UnauthenticatedPageContent in App.jsx

yonidoronpeters committed:
DHP-277 Add and fix tests for "Sign in or create an account" button

yonidoronpeters committed:
DHP-277 Small styling fix
